src/system-health/scripts/healthd

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python2
+
+"""
+    healthd
+    System health monitor daemon for SONiC
+"""
+
+import signal
+import threading
+
+from sonic_py_common.daemon_base import DaemonBase
+from swsssdk import SonicV2Connector
+
+from health_checker.manager import HealthCheckerManager
+
+SYSLOG_IDENTIFIER = 'healthd'
+
+
+class HealthDaemon(DaemonBase):
+    """
+    A daemon that run as a service to perform system health checker with a configurable interval. Also set system LED 
+    according to the check result and store the check result to redis.
+    """
+    SYSTEM_HEALTH_TABLE_NAME = 'SYSTEM_HEALTH_INFO'
+
+    def __init__(self):
+        """
+        Constructor of HealthDaemon.
+        """
+        DaemonBase.__init__(self, SYSLOG_IDENTIFIER)
+        self._db = SonicV2Connector(host="127.0.0.1")
+        self._db.connect(self._db.STATE_DB)
+        self.stop_event = threading.Event()
+
+    def deinit(self):
+        """
+        Destructor. Remove all entries in $SYSTEM_HEALTH_TABLE_NAME table.
+        :return: 
+        """
+        self._clear_system_health_table()
+
+    def _clear_system_health_table(self):
+        self._db.delete_all_by_pattern(self._db.STATE_DB, HealthDaemon.SYSTEM_HEALTH_TABLE_NAME)
+
+    # Signal handler
+    def signal_handler(self, sig, frame):
+        """
+        Signal handler
+        :param sig: Signal number
+        :param frame: not used
+        :return:
+        """
+        if sig == signal.SIGHUP:
+            self.log_info("Caught SIGHUP - ignoring...")
+        elif sig == signal.SIGINT:
+            self.log_info("Caught SIGINT - exiting...")
+            self.stop_event.set()
+        elif sig == signal.SIGTERM:
+            self.log_info("Caught SIGTERM - exiting...")
+            self.stop_event.set()
+        else:
+            self.log_warning("Caught unhandled signal '" + sig + "'")
+
+    def run(self):
+        """
+        Check system health in an infinite loop.
+        :return: 
+        """
+        self.log_info("Starting up...")
```

Comment:
Note that the default log priority of the Logger class is currently NOTICE. If you want these INFO level logs to get logged, you will need to set the minimum log priority to INFO using `set_min_log_priority_info()` like is done here: https://github.com/Azure/sonic-buildimage/pull/5370/files#diff-fd5407b6f3a90e41c17bb90532f90bd1R218